src/gtk/settings.cpp

```diff
@@ -840,6 +853,53 @@ wxFont wxSystemSettingsNative::GetFont( wxSystemFont index )
     return font;
 }
 
+#if defined(__WXGTK3__) || defined(__WXGTK__)
+bool wxSystemAppearance::IsSystemDark() const
+{
+    GVariant *out = nullptr;
+    GVariant *child = nullptr;
+    GVariant *ret = nullptr;
+    GError *error = nullptr;
+
+    bool isDark = IsDark();
+
+    GDBusProxy *proxy = g_dbus_proxy_new_for_bus_sync(G_BUS_TYPE_SESSION,
+        G_DBUS_PROXY_FLAGS_NONE,
+        nullptr,
+        PORTAL_BUS_NAME,
+        PORTAL_OBJECT_PATH,
+        PORTAL_SETTINGS_INTERFACE,
+        nullptr,
+        &error);
+
+    if (proxy) {
+        ret = g_dbus_proxy_call_sync(proxy, "Read",
+            g_variant_new ("(ss)", "org.freedesktop.appearance", "color-scheme"),
+            G_DBUS_CALL_FLAGS_NONE, G_MAXINT, nullptr, nullptr);
+
+        if (ret) {
+            g_variant_get(ret, "(v)", &child);
+            g_variant_get(child, "v", &out);
+
+            isDark = g_variant_get_uint32(out) == FDO_PREFER_DARK;
```

Comment:
I'm not 100% sure if we should set it to false for `FDO_DEFAULT`, it could be better to fallback to `IsDark()` in this case.